sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -903,6 +904,28 @@ def test_non_euclidean_kneighbors():
                   radius, metric='euclidean')
 
 
+def test_kneighbors_parallel():
+    X = iris.data
+    y = iris.target
+    X_train, X_test, y_train, y_test = train_test_split(X, y)
+    for algorithm in ALGORITHMS:
+        clf = neighbors.KNeighborsClassifier(n_neighbors=3,
+                                             algorithm=algorithm)
+        clf.fit(X_train, y_train)
+        y_1 = clf.predict(X_test)
+        dist_1, ind_1 = clf.kneighbors(X_test)
+        A_1 = clf.kneighbors_graph(X_test, mode='distance').todense()
+        for n_jobs in [2]:
```

Comment:
In this case 2 *threads* (in the same process) run concurrently on the same core (the OS will switch) them. Modern OS have no problem running a large amount (tens of thousands) of threads and process on a single core. It's just no bringing any speed benefit to running the same tasks sequentially.